dace/transformation/passes/dead_state_elimination.py

```diff
@@ -137,13 +195,15 @@ def is_definitely_taken(self, edge: InterstateEdge, sdfg: SDFG) -> bool:
             return True
 
         # Evaluate condition
-        scond = edge.condition_sympy()
-        if scond == True or scond == sp.Not(sp.logic.boolalg.BooleanFalse(), evaluate=False):
+        return self._is_truthy(edge.condition_sympy(), sdfg)
+
+    def _is_truthy(self, cond: sp.Basic, sdfg: SDFG) -> bool:
```

Comment:
I really don't like the names is_truthy and is_falsy. Sounds weird. Why not `_is_definitely_true` or something like that?